ea.go

```diff
@@ -28,6 +37,44 @@ type ExtendedAttribute struct {
 	Flags uint8
 }
 
+// GetFileEA retrieves the extended attributes for the file represented by `handle`. The
+// `handle` must have been opened with file access flag FILE_READ_EA (0x8).
+func GetFileEA(handle windows.Handle) ([]ExtendedAttribute, error) {
+	// default buffer size to start with
+	bufLen := 1024
+	buf := make([]byte, bufLen)
+	var iosb ioStatusBlock
+	// keep increasing the buffer size until it is large enough
+	for {
+		status := getFileEA(handle, &iosb, &buf[0], uint32(bufLen), false, 0, 0, nil, true)
+		if status.Err() != nil {
```

Comment:
Thank you for this contribution @ambarve and I'm thankful to all the contributors of this project, it's been very helpful.

I think one aspect that could be improved in this function is that if you pass a handle to a file which does not have any extended attributes, then it returns an error. However, this should ideally be a normal situation that should be handled gracefully.
I noticed that the system call returns -1073741742 status code for both files and directories in this case.

Suggested change -
for { status := getFileEA(handle, &iosb, &buf[0], uint32(bufLen), false, 0, 0, nil, true) if status == -1073741742 { //If status is -1073741742, no extended attributes were found return nil, nil } else { err := status.Err() if err != nil { // convert ntstatus code to windows error if err == windows.ERROR_INSUFFICIENT_BUFFER || err == windows.ERROR_MORE_DATA { bufLen *= 2 buf = make([]byte, bufLen) continue } return nil, fmt.Errorf("get file EA failed with: %w", err) } } break }

Said that, I do not know if this is the best way to handle this scenario or if that status code will always be consistent.
Anyone with a more comprehensive understanding of the system internals have any better suggestions?